android/libraries/rib-android/src/main/java/com/uber/rib/core/RibActivity.kt

```diff
@@ -0,0 +1,58 @@
+package com.uber.rib.core
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.ViewGroup
+
+abstract class RibActivity : AppCompatActivity() {
+
+    private lateinit var rootNode: Node<*>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        rootNode = createRib().apply {
+            dispatchAttach(savedInstanceState)
+            attachToView(rootViewGroup)
+        }
+    }
+
+    abstract val rootViewGroup: ViewGroup
+
+    abstract fun createRib(): Node<*>
+
+    override fun onStart() {
```

Comment:
Did you discuss this approach with getting activityLifeCycle events?
What should we do if we will need to get another event? For example: onPictureInPictureModeChanged

Reply:
We will add those as well. If it gets too much, we can revert to Uber's implementation of abstracting these events and callbacks.